packages/system/src/generator.ts

```diff
@@ -50,40 +50,46 @@ export class StyleGenerator {
         !/^\w+\(.*\)$/.test(propValue) // exclude CSS functions
       );
     };
-
-    for (const [propName, propValue] of Object.entries(props)) {
+    /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-return -- FIXME */
+    const convertStyle = (name: string, value: any) => {
       // color={['colors.primary', 'colors.secondary']}
-      if (Array.isArray(propValue)) {
-        styledProps[propName] = propValue.map((value) => {
-          if (isThemeStyle(value)) {
-            const customStyle = findThemeStyle(value);
+      if (Array.isArray(value)) {
+        return value.map((v) => {
+          if (isThemeStyle(v)) {
+            const customStyle = findThemeStyle(v);
             if (customStyle !== undefined) {
               return customStyle;
             }
           }
-          return value;
+          return v;
         });
-      }
-      // color="colors.primary"
-      else if (isThemeStyle(propValue)) {
-        const customStyle = findThemeStyle(propValue);
+        // color="colors.primary"
+      } else if (isThemeStyle(value)) {
+        const customStyle = findThemeStyle(value);
         if (customStyle !== undefined) {
-          styledProps[propName] = customStyle;
+          return customStyle;
         }
-      } else if (isStyledProp(propName)) {
-        styledProps[propName] = propValue;
-      } else if (isPseudoProps(propName)) {
+      } else if (isStyledProp(name)) {
+        return value;
+      }
+      return value;
```

Comment:
Like I have commented out examples of return values in other lines (for example: L55 & L66), it would be helpful if you could add a comment here as well indicating what kind of values can be returned